ipalib/x509.py

```diff
@@ -405,6 +405,18 @@ def match_hostname(self, hostname):
     def tbs_precertificate_bytes(self):
         return self._cert.tbs_precertificate_bytes
 
+    # added in python-cryptography 40.0
+    def verify_directly_issued_by(self, issuer):
+        """
+        This method verifies that certificate issuer name matches the
+        issuer subject name and that the certificate is signed by the
+        issuer's private key. No other validation is performed.
+        """
+        if 'verify_directly_issued_by' in self._cert.__abstract_methods__:
+            return self._cert.verify_directly_issued_by(issuer)
+        else:
+            return None
+
```

Comment:
This approach defines the additional concrete method if and only if the ABC has the new method.

```suggestion
    if hasattr(crypto_x509, "verify_directly_issued_by"):
        # added in python-cryptography 40.0
        def verify_directly_issued_by(self, issuer):
            return self._cert.verify_directly_issued_by(issuer)
```

Reply:
typo:
`crypto_x509` => `crypto_x509.Certificate`.